### DISCUSSION TITLE
just curious..

### DISCUSSION THREAD

**Discussion by dobkeratops** [trunkmaster/nextspace#391] [Ideas]
great to see this..

how far can it go with those 'tear off menus' - could it be wired up such that any 'global-menu' friendly applications use those distinctive nextstep tear off menus.. or do applications have to specifically be adapted to use them

**Comment by alexmyczko** [comment]
GNUstep applications have that by default. Nothing needed to do by the developer. Forget about gtk and qt.

GNUstep also has theming with Windows and macOS style menus, all automatically built-in, here's some screenshots.
https://gnutopics.wordpress.com/2014/06/06/theming-and-style-in-gnustep/
https://github.com/BertrandDekoninck/rik.theme/blob/master/newscreen.png

**Comment by dobkeratops** [comment]
i imagine there's a theoretical possibility to make a global-menu implementation for gtk/qt that would look like gnustep/nextstep menus... but perhaps it's an insane amount of work just for a nostalgic itch. 

I'd like it.. but I probably wouldn't volunteer the time myself -  and of course I dont know for sure that it's possible.. I haven't looked into it. I think they had problems with globalmenu/gnome shell?

**Comment by alexmyczko** [comment]
my point was I'm fully with GNUstep, since they have the inter application thing called Services (also available on iOS, macOS). a great thing. (define word, in DictionaryReader on selected word in TextEdit).
So i wouldn't want gtk/qt extended with it, since key combinations (copy paste) consistency will break.

And then there's the font dialog with kerning, which is also only done right in GNUstep and macOS/iOS, and not in qt, nor gtk.

And don't even get me started with the file open/save dialogs...

List goes on here: https://en.wikipedia.org/wiki/GNUstep#Class_capabilities
gtk was made for gimp, for linux. Guess why it took ages for a half-nice Windows, macOS version.
GNUstep (well OpenStep specs considered portability among all kinds of systems from the early beginning on)

**Comment by trunkmaster** [comment]
@dobkeratops, I think it's possible. AFAIK it should be a D-Bus messages receiver from global menu-aware application. I've already thought about it... But if we speak about GTK/Qt applications I'd rather create GNUstep versions of Transmission, VLC or something that already has MacOS native port.

Another task I'm thinking about is a do something with systray icons. Think about Skype, Dropbox etc...
I'm taking about design point of view. I even tried to create mockup of floating titlebar style systtray...

**Comment by ghost** [comment]
Once GTK and QT have working global menu panels, someone could theoretically theme them to look and behave like a NeXT menu...
As for making a panel in GNUstep that captures GTK and QT apps, that would be a little bit more involved.
